motioneye/uploadservices.py

```diff
@@ -381,9 +381,9 @@ def upload_data(self, filename, mime_type, data, ctime, camera_name):
             '',
         ]
 
-        body = '\r\n'.join(body)
+        body = '\r\n'.join(body).encode()
         body += data
-        body += f'\r\n--{self.BOUNDARY}--'
+        body += f'\r\n--{self.BOUNDARY}--'.encode()
```

Comment:
What do your think about doing the encoding consistently once when passing it to the request? So once doesn't need to take care of this when changing the code which crafts the body. In your case, is `data` bytes already? I wonder how it was even possible before to concatenate it to a string and another string afterwards. However, if possible, better to encode the whole body in one turn instead doing this for every part separately.

Reply:
I do not have the skill to be done in a single call.